orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/RedisConfiguration.groovy

```diff
@@ -66,6 +66,20 @@ class RedisConfiguration {
     return createPool(null, previousConnection, timeout, registry, "jedisPoolPrevious")
   }
 
+  @Bean(name="redisClientDelegate")
+  @Primary
+  RedisClientDelegate redisClientDelegate(@Qualifier("jedisPool") Pool<Jedis> jedisPool) {
+    return new JedisClientDelegate(jedisPool)
+  }
+
+  @Bean(name="previousRedisClientDelegate")
+  RedisClientDelegate previousRedisClientDelegate(@Qualifier("jedisPoolPrevious") JedisPool jedisPoolPrevious) {
+    if (jedisPoolPrevious == null) {
+      return null
+    }
```

Comment:
I think this might be better done using `@ConditionalOnBean`. I'm not sure what the implications are of returning `null` from a `@Bean` method.

Reply:
done, similarly refactoring `jedisPoolPrevious()` in the same class